app/views/smart_proxies/plugins/_puppet.html.erb

```diff
@@ -0,0 +1,29 @@
+<% dashboard = Dashboard::Data.new("puppetmaster = #{@smart_proxy.name}") %>
+<% @hosts = dashboard.hosts %>
+<% @report = dashboard.report %>
+<h3><%= _("Puppet") %></h3>
+<%= show_feature_version('puppet') %>
+<div class="col-md-2"><%= _('Hosts managed:') %></div>
+<div class="col-md-10">
+  <%= link_to @hosts.length, hosts_path(:search => "puppetmaster = #{@smart_proxy.name}") %>
+</div>
+<ul id="proxy-puppet-tab" class="nav nav-tabs">
+  <li class="active"><a href="#puppet_dashboard" data-toggle="tab"><%= _("Statistics")%></a></li>
+  <li><a href="#puppet_envs" data-toggle="tab"><%= _('Environments') %></a></li>
+</ul>
+<div class="tab-content proxy-content" id="proxy-puppet-tab-content">
+  <div class="tab-pane active in" id="puppet_dashboard">
+    <div class="col-md-5">
+      <%= render :partial => 'dashboard/reports_widget' %>
```

Comment:
I think this makes sense, though I think there was a reason I used instance variables in the original dashboard... this can be refactored separately in any case, not to complicate this PR even more.